lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1438,6 +1452,14 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     }
     if (!Scope)
       Scope = getOrCreateContext(Context);
+
+    // Scope outermost fileprivate decls in an inline private discriminator
+    // namespace.
+    if (auto *Decl = DbgTy.getDecl())
+      if (Context && !Decl->isInPrivateOrLocalContext() &&
+          Decl->isOutermostPrivateOrFilePrivateScope())
```

Comment:
Now you don't need the first condition, because it's folded into the second.

Reply:
Argh — I pushed before amending my commit. I knew this one looked oddly familiar.